mne/io/reference.py

```diff
@@ -197,6 +197,9 @@ def add_reference_channels(inst, ref_channels, copy=True):
     else:  # n_ref_channels == 1 and a single ref digitization exists
         ref_dig_array = np.concatenate((ref_dig_loc[0]['r'],
                                        ref_dig_loc[0]['r'], np.zeros(6)))
+        # Replace the (possibly new) Ref location for each channel
```

Comment:
Found it... Should by no means postpone this PR, but:

The fwd code is too nested for me to figure it out. I cannot wrap my head around why the physical location of the Ref would need to enter the forward computation. As long as the reference used was proper, the CAR obviates the need, I thought. Under CAR, _any_ measurement configuration should be identical to the "infinite potential" solutions obtained using a forward model, once that is also CAR'ed (assumptions: perfectly accurate head model & electrode locations...). There are certain "risks" of modeling the Ref explicitly (_e.g._, tip of nose with defaced MRI), and I had convinced myself of the "fact" that it was unnecessary. And as I write this, I begin to doubt again...